Services/Services.Application/Services/DoctorService.cs

```diff
@@ -0,0 +1,69 @@
+﻿using AutoMapper;
+using Services.Application.Models;
+using Services.Domain.Repositories.Abstractions;
+using Services.Application.Services.Abstractions;
+using Services.Domain.Entities;
+using Services.Domain.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Services.Application.Services
+{
+    public class DoctorService : IDoctorService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public DoctorService(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task Create(CreateDoctorModel createModel, CancellationToken cancellationToken)
+        {
+            // Get related to doctor specialization
+            var specializationRelatedToDoctor = await _unitOfWork.GetSpecializationRepository().GetAsync(createModel.SpecializationId, cancellationToken);
+            // If specified specialization is not active or not found, can't create entity
+            if (specializationRelatedToDoctor == null || !specializationRelatedToDoctor.IsActive)
+            {
+                throw new RelatedObjectNotFoundException($"Related specialization with id {createModel.SpecializationId} is not found or not active.");
+            }
+            await _unitOfWork.GetDoctorRepository().CreateAsync(_mapper.Map<Doctor>(createModel), cancellationToken);
+            await _unitOfWork.SaveChangesAsync(cancellationToken);
+        }
+
+        public async Task Delete(Guid id, CancellationToken cancellationToken)
+        {
+            await _unitOfWork.GetDoctorRepository().DeleteAsync(id, cancellationToken);
```

Comment:
What will happen if doctor with target id not exists in DB?

Reply:
I am going to create NotFoundException in infrastucture layer, and if doctor with this id doesnt exist, exception will be thrown, is it good?

Reply:
I want to create exception (NotFoundException) at infrastructure layer, and throw it when entity is not found in repositories, is it ok?

Reply:
or create exception in domain layer, because repositoriy interfaces in domain layer

Reply:
Fixed